### PR TITLE
obspy.earthworm client times out on Py3.

### PR DESCRIPTION
**Pull request by QuLogic**
Okay this one is a follow up on  #989, which fixed Py2 behavior. Py3 still fails with a timeout:
```
petr% python3.4 ./test-ew.py
socket timeout in getSockCharLine()
socket timeout in getSockCharLine()
 ```
I also checked the public server I indicated earlier. It should be available on `pubavo1.wr.usgs.gov:16022`. A minimal connection test can be performed with telnet, the request `MENU: 0` should provide a sort of inventory. This is what I obtain:
```
petr% telnet pubavo1.wr.usgs.gov 16022
Trying 130.118.181.39...
Connected to pubavo1.wr.usgs.gov.
Escape character is '^]'.
MENU: 0
  239 ACH EHE AV 1421685777.065 1426869775.0500011 s4  76 ACH EHN AV 1421685777.065 1426869774.0500011 s4  74 ACH EHZ AV 1421685777.065 1
[...]
```

### CODE REVIEW

**Comment by QuLogic** [comment]
@petrrr Try that one your internal server. I guess the public one is not completely standard, though.

**Comment by mbyt** [comment]
@QuLogic do you mind to give me push access to `obspy/QuLogic`.

Otherwise, I need to go for the complicated procedure of forking the pull request, do some changes and propose a new pull request, which I like to avoid...

**Comment by mbyt** [comment]
@petrrr should work now.

File: CHANGELOG.txt [hunk 1]
```diff
@@ -1,3 +1,7 @@
+development:
```
Comment (QuLogic) [inline CHANGELOG.txt]: This is on releases, so it'd be better to mark it 0.10.x just to ease merging (if we don't make any more point releases, that is.)

**Comment by petrrr** [comment]
Thanks a lot for your work. Unfortunately, I cannot confirm definitely this solved the issue we observed with our server until returning to the office.

**Comment by petrrr** [comment]
Okay! I now can confirm that this also now works fine with our server. So it is probably good to merge. BTW: could this make its way into a bugfix release?

**Comment by megies** [comment]
> could this make its way into a bugfix release?

We'll merge it into releases. But it depends on whether we decide it's worth to do a 0.10.2 or head straight for next major release.

**Comment by megies** [comment]
Felt free to rebase and force-push for easy merging..

**Comment by petrrr** [comment]
@megies, as far as I understand this PR is based on release and could be applied directly without rebase. It will need to be rebased to be applied to master with the new directory layout.

**Comment by petrrr** [comment]
Closing this in favor of the rebased PR  #1007.

**Comment by QuLogic** [comment]
I'll merge this when CI passes. The main change is already okay, it's just docs and polish that were added.

File: obspy/earthworm/client.py [hunk 2]
```diff
@@ -77,22 +77,22 @@ def getWaveform(self, network, station, location, channel, starttime,
...
             from obspy import UTCDateTime
-            client = Client("pele.ess.washington.edu", 16017, timeout=5)
-            dt = UTCDateTime(2013, 1, 17) - 2000  # now - 2000 seconds
-            st = client.getWaveform('UW', 'TUCA', '', 'BHZ', dt, dt + 10)
+            client = Client("pubavo1.wr.usgs.gov", 16022, timeout=5)
+            dt = UTCDateTime() - 2000  # now - 2000 seconds
+            st = client.getWaveform('AV', 'ACH', '--', 'EHE', dt, dt + 10)
             st.plot()
-            st = client.getWaveform('UW', 'TUCA', '', 'BH*', dt, dt + 10)
+            st = client.getWaveform('AV', 'ACH', '--', 'EH*', dt, dt + 10)
```
Comment (krischer) [inline obspy/earthworm/client.py]: I did not follow this PR in detail, but why is the `--` needed? All other clients treat (or at least should treat) an empty string as an empty location code. There was also a LONG thread on the FDSN mailing list about this issue a couple months back. Why not just internally map it to dashes?

Comment (QuLogic) [inline obspy/earthworm/client.py]: It looks like `getWaveform` does handle empty strings, but `availability` doesn't (which is why I thought it was necessary.)

Comment (krischer) [inline obspy/earthworm/client.py]: All of them should probably be made to handle empty strings IMHO. I don't mind if they also work with dashes, but at least the tests and docs should use empty strings. Then it is consistent with the other clients.

**Comment by petrrr** [comment]
@krischer wrt the `--`: if I understood correctly how the client is implemented (and we do not want to touch this) it will be necessary. Basically the availability requests the whole "menu". If there is no locID, it will be set to `--`. Then the whole list is matched for the pattern defined by the parameters to the `availabilty()` method.

**Comment by krischer** [comment]
so `--` internally is like a wildcard? That's a bit strange to me as `*` appears to be the wildcard for the channel codes.

Could we just have the following ruleset for the outwards facing `Client.getWaveform()` method:

1. `location=None`: Wildcard; download data with any location code.
2. `location=""`: Explicitly empty location code.
3. `location="--"`: Raise?

**Comment by QuLogic** [comment]
> If there is no locID, it will be set to --.

Doesn't seem to be a wildcard to me. Sounds like it works like an empty location, just like you asked about.

**Comment by krischer** [comment]
So this could just be replaced by an empty string and work the same?

**Comment by QuLogic** [comment]
Seems to work: 9bcfc87. Only thing is the result would not be empty, but `--`.
